Sources/NIO/ChannelPipeline.swift

```diff
@@ -1017,11 +1017,29 @@ public final class ChannelHandlerContext: ChannelInvoker {
     }
 
     public var remoteAddress: SocketAddress? {
-        return try? self.channel._unsafe.remoteAddress0()
+        do {
+            // Fast-path access to the remoteAddress.
+            return try self.channel._unsafe.remoteAddress0()
+        } catch let err as ChannelError where err == .ioOnClosedChannel {
```

Comment:
No need for this verbose syntax, just do `catch ChannelError.ioOnClosedChannel`.